lib/src/async.dart

```diff
@@ -29,14 +29,18 @@ class _FutureStateHook<T> extends HookState<AsyncSnapshot<T>, _FutureHook<T>> {
   /// An object that identifies the currently active callbacks. Used to avoid
   /// calling setState from stale callbacks, e.g. after disposal of this state,
   /// or after widget reconfiguration to a new Future.
-  Object _activeCallbackIdentity;
-  AsyncSnapshot<T> _snapshot;
+  Object? _activeCallbackIdentity;
+  late AsyncSnapshot<T> _snapshot;
```

Comment:
Prefer nullable properties instead of `late`.

That keyword is dangerous

Reply:
So would you want useFuture to return a nullable AsyncSnaphot (`AsyncSnapshot<T>? useFuture<T>`) or should the build method return a non nullable snapshot since it should be initialized anyway?

Reply:
How would you feel about
`late AsyncSnapshot<T> _summary = AsyncSnapshot<T>.withData(ConnectionState.none, hook.initialData);`?

Reply:
The return value must be non-nullable



> How would you feel about
> `late AsyncSnapshot<T> _summary = AsyncSnapshot<T>.withData(ConnectionState.none, hook.initialData);`?

That's fine, it's mostly uninitialized late variables that are scary

Then we can do the same thing with other hooks, like useStreamController